cloud/ec2_util.go

```diff
@@ -299,12 +299,35 @@ type Terms struct {
 	}
 }
 
-// format /dev/sd[f-p][1-6] taken from https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html
-func generateDeviceNameForVolume() string {
+func getGeneratedDeviceNameForVolume(ctx context.Context, isWindowsHost bool) (string, error) {
+	deviceName := ""
+	err := util.Retry(
+		ctx,
+		func() (bool, error) {
+			deviceName = generateDeviceNameForVolume(isWindowsHost)
+			exists, err := host.HostExistsWithVolumeWithDeviceName(deviceName)
+			if err != nil {
+				return true, errors.Wrapf(err, "error checking if device name already exists")
+			}
+			if !exists {
+				return false, nil
+			}
+			return true, errors.New("generated device name already exists")
+		}, 500, 0, 0)
```

Comment:
I'm not sure what passing 0 for min/max duration are meant to do. Looking at our code, we start at 100ms sleeps (overriding shorter set values for min) and when max is 0 we do (min)*2^<attempts>, which is 500, so five seconds? is that what we want?

Reply:
Time between retries for this wouldn't matter, so no. I'll change it.